test/infrastructure/docker/e2e/docker_suite_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/generators"
+
+	corev1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha2"
+	infrav1 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1alpha2"
+)
+
+func TestDocker(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Docker Suite")
+}
+
+var mgmt *CAPDCluster
+var ctx = context.Background()
+
+var _ = BeforeSuite(func() {
+	managerImage := os.Getenv("MANAGER_IMAGE")
+	if managerImage == "" {
+		managerImage = "gcr.io/k8s-staging-capi-docker/capd-manager-amd64:dev"
+	}
+	By("setting up in BeforeSuite")
+	var err error
+
+	// Set up the provider component generators based on master
+	core := &generators.ClusterAPI{GitRef: "master"}
+
+	// set up capd components based on current files
+	infra := &provider{}
+
+	// set up cert manager
+	cm := &generators.CertManager{ReleaseVersion: "v0.11.0"}
```

Comment:
Should we be able to default this with the CertManager generator? I suspect most cases providers would want to default to the same version that is used by the vendored version of CAPI.

Reply:
It would be really nice to be able pull in the required version automatically. Did you have any ideas for how we might approach that? I couldn't even come up with anything that exists today as a solution.

This could be a future improvement on the framework.